gerrychain/partition/partition.py

```diff
@@ -58,7 +59,11 @@ def _first_time(self, graph, assignment, updaters):
         if updaters is None:
             updaters = dict()
 
-        self.updaters = self.default_updaters.copy()
+        if use_cut_edges:
+            self.updaters = {"cut_edges": cut_edges}
+        else:
+            self.updaters = {}
+
         self.updaters.update(updaters)
```

Comment:
I think tackling #370 here makes sense: if we're modifying how dictionaries of updaters are stored/passed around, we might as well make the necessary `UserWarning` + deepcopy change.

Reply:
Sure, but I think that'd fit in better as a separate PR (since we should add some tests to make sure that `deepcopy` prevents #370 and a `UserWarning` is thrown at appropriate times)